### PR TITLE
EASY-2524: validate DOI and URN related identifiers make a valid URL

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-2524

#### When applied it will
* validate that DOI and URN related identifiers make a valid URL

image

#### To be discussed before merging
* [x] I don't see XML-schema validation for these related identifiers in terms of "are these valid identifiers". However, in `easy-validate-dans-bag` there is some validation for DOIs (https://github.com/DANS-KNAW/easy-validate-dans-bag/blob/ad8977298b4ea464e4b7c87708b2bb4a1df6c3bc/src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala#L109-L131). @lindareijnhoudt should we add that here as well?

@DANS-KNAW/easy for review

### CODE REVIEW

**Comment by lindareijnhoudt** [comment]
yes, in order to make sure submitted deposits are not later on rejected on preventable causes. for the DOI, this check is a regexp. what do you expect for the urn value?

File: src/main/resources/constants/identifiers.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
     "title": "doi",
-    "viewName": "doi"
+    "viewName": "doi",
+    "baseUrl": "https://doi.org/"
   },
   "id-type:URN": {
     "title": "urn",
-    "viewName": "urn"
+    "viewName": "urn",
+    "baseUrl": "http://persistent-identifier.nl/"
```
Comment (lindareijnhoudt) [inline src/main/resources/constants/identifiers.json]: this holds only for the dutch URN-NBN, this field is for any URN, which is broader

Comment (rvanheest) [reply src/main/resources/constants/identifiers.json]: Then we got a problem in here: https://github.com/DANS-KNAW/easy-deposit-api/blob/30f196bf9d32e10e14981007531f21e333ff69fe/src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala#L109

Probably we should fix that one first if this URN field is not just for our URN-NBN...

Comment (rvanheest) [reply src/main/resources/constants/identifiers.json]: we decided to make it more explicit to make `viewName: urn:nbn:nl` and add format checking to URN (`^urn:nbn:nl:ui:.*$`) and DOI (`^(http://dx.doi.org/|https://doi.org/){0,1}10.(\d)+(\.(\d)+)*/.*$`)

**Pull request merged**